connectivity/drivers/ble/FEATURE_BLE/COMPONENT_BlueNRG_MS/CMakeLists.txt

```diff
@@ -1,12 +1,9 @@
 # Copyright (c) 2020 ARM Limited. All rights reserved.
 # SPDX-License-Identifier: Apache-2.0
 
-target_sources(mbed-ble-blue_nrg
-    INTERFACE
-        BlueNrgMsHCIDriver.cpp
-)
+add_library(mbed-ble-bluenrg_ms INTERFACE)
```

Comment:
How do we make the decision which library should be `INTERFACE`?

Reply:
if CMake would allow circular deps, this would be OBJECT library.

This library provides functionality declared in BLE.
To overcome this limitation, the main library that is user facing lib (mbed-ble in this case) is OBJECT library. Extensions like components are INTERFACES as they can't built without the library that declares the functionality.